docs/add-runtime.md

```diff
@@ -0,0 +1,254 @@
+
+Finally, you will need to create a `feat-XXX-YYY-runtime` branch based on the `refactor` branch and switch to it. The `XXX` should represent the issue ID and `YYY` the runtime name.
+
+## 2. Implement new runtime
+
+### 2.1 Preparing the files for your new runtime
+The first step to writing a new runtime is to create a folder within `/runtimes` with the name of the runtime and the version separated by a dash. For instance, if I was to write a Rust Runtime with version 1.55 the folder name would be: `rust-1.55`
+
+Within that folder you will need to create a few basic files that all OpenRuntimes runtimes require:
+```
+Dockerfile - Dockerfile that explains how the container will be built.
+README.md - A readme file explaining the runtime and any special notes for the runtime. A good example of this is the PHP 8.0 runtime.
+```
+
+### 2.2 Differences between compiled and interpreted runtimes
+Runtimes within OpenRuntimes are created differently depending on whether they are compiled or interpreted. This is due to the fundamental differences between the two ways of running the code.
+
+Interpreted languages have both a `build.sh` file and a `launch.sh` file.
+The `build.sh` file for an interpreted runtime is normally used for installing any dependencies for both the server itself and the user's code and then to copy it to the `/usr/code` folder which is then packaged and can be used later for running the server.
+The build script is always executed during the build stage of tag deployment.
+
+The `launch.sh` file for an interpreted runtime should extract the `/tmp/code.tar.gz` file that contains both the user's code and the dependencies. This tarball was created by OpenRuntimes from the `/usr/code` folder and should install the dependencies that were pre-installed by the build stage and move them into the relevant locations for that runtime. It will then run the server ready for execution.
+
+---
+Compiled Languages only have a `build.sh` file.
+The `build.sh` script for a compiled runtime is used to move the user's source code and rename it into source files for the runtime (The `ENTRYPOINT_NAME` environment variable can help with this) it will also build the code and move it into the `/usr/code` folder. Compiled runtime executables **must** be called `runtime` for the ubuntu or alpine images to detect and run them.
+
+#### Note:
+`/tmp/code.tar.gz` is always created from the `/usr/code` folder in the build stage. If you need any files for either compiled or interpreted runtimes you should place them there and extract them from the `/tmp/code.tar.gz` during the `launch.sh` script to get the files you need.
+
+### 2.3 Writing the runtime
+Internally the runtime can be anything you like as long as it follows the standards set by the other runtimes.
+
+The best way to go about writing a runtime is like so:
+Initialize a web server that runs on port 3000 and uses any IP Address (0.0.0.0) and on each `POST` request do the following:
```

Comment:
```suggestion
Initialize a web server that runs on port 3000 and uses any IP Address (0.0.0.0) and on each `POST` request does the following:
```
I also dont think its okay to use any IP address. It has to be `0.0.0.0` I believe

Reply:
I'm not sure using "does" here is grammatically correct, it seems out of place reading it in my head

Reply:
Oh my bad. Ignore the `does` comment. 👍  Check out the one about the server IP

Reply:
alright, addressed 👍